include/seqan3/range/view/translation.hpp

```diff
@@ -45,292 +45,657 @@
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator begin() const noexcept
+    {
+        return {*this, 0};
+    }
+
+    //!\copydoc begin()
+    iterator cbegin() const noexcept
+    {
+        return begin();
+    }
+
+    /*!\brief Returns an iterator to the element following the last element of the container.
+     * \returns Iterator to the first element.
+     *
+     * This element acts as a placeholder; attempting to dereference it results in undefined behaviour.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator end() const noexcept
+    {
+        return {*this, size()};
+    }
+
+    //!\copydoc end()
+    iterator cend() const noexcept
+    {
+        return end();
+    }
+    //!\}
+
+     /*!\brief Returns the number of elements in the view.
+     * \returns The number of elements in the container.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (never modifies data).
+     */
+    size_type size() const
+    {
+        switch (data_members->tf)
         {
-            return irange  | ranges::view::chunk(3)
-                           | ranges::view::take_while([] (auto chnk) { return ranges::size(chnk) == 3; })
-                           | ranges::view::transform(translate);
+            case translation_frames::FWD_FRAME_0:
+                [[fallthrough]];
+            case translation_frames::REV_FRAME_0:
+                return ranges::size(data_members->urange) / 3;
+                break;
+            case translation_frames::FWD_FRAME_1:
+                [[fallthrough]];
+            case translation_frames::REV_FRAME_1:
+                return (ranges::size(data_members->urange) - 1) / 3;
+                break;
+            case translation_frames::FWD_FRAME_2:
+                [[fallthrough]];
+            case translation_frames::REV_FRAME_2:
+                return (ranges::size(data_members->urange) - 2) / 3;
+                break;
+            default:
+                throw std::invalid_argument(multiple_frame_error);
+                break;
         }
     }
 
-    /*!\brief Pipe operator that enables view-typical use of pipe notation.
-    * \tparam irng_t The type of the range being processed.
-    * \param irange The range being processed as left argument of the pipe.
-    * \param translate_single_fn_v The result of the single-argument operator().
-    */
-    template <typename irng_t>
-    friend auto operator|(irng_t && irange, seqan3::detail::translate_single_fn const & translate_single_fn_v)
+    /*!\name Element access
+     * \{
+     */
+    /*!\brief Return the n-th element.
+     * \param n The element to retrieve.
+     *
+     * Accessing an element behind the last causes undefined behaviour. In debug mode an assertion checks the size of
+     * the container.
+     *
```

Comment:
please add a Complexity section here, especially for random access this is important in our library.